BSP/Src/BSP_UART.c

```diff
@@ -0,0 +1,87 @@
+#include "BSP_UART.h"
+
+static USART_TypeDef *handles[NUM_UART] = {USART1, USART2, USART3};
+
+static UART_HandleTypeDef for_interrupts;
+
+//TODO This function needs to be reviewed, and queue creation will likely happen here
+BSP_Status BSP_UART_Init(UART_t device) {
```

Comment:
I didn't review function implementations yet since that's very subject to change, only skeletons. Once you are done we will review implementations.